### PR TITLE
Add FileManager actions in CloudStorage

### LINKED ISSUE #163: Add FileManager actions in CloudStorage

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Closes #163

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/data/documentprovider/CloudStorageProvider.kt [hunk 1]
```diff
@@ -327,29 +561,50 @@ class CloudStorageProvider : DocumentsProvider() {
...
         var flags = 0
 
-        if (file?.isFolder() == true) flags = flags or DocumentsContract.Document.FLAG_DIR_SUPPORTS_CREATE
+        if ((file?.isFolder() == true && file.rights?.newFile == true) || isRootFolder) {
+            flags = flags or DocumentsContract.Document.FLAG_DIR_SUPPORTS_CREATE
+        }
 
-        val mimetype = if (file == null || file.isFolder()) DocumentsContract.Document.MIME_TYPE_DIR else file.getMimeType()
+        val mimetype = if (file == null || file.isFolder()) {
+            // flags = flags or DocumentsContract.Document.FLAG_DIR_SUPPORTS_CREATE
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/data/documentprovider/CloudStorageProvider.kt:589]: Why this commented code?

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/utils/Extensions.kt [hunk 2]
```diff
@@ -757,3 +759,15 @@ fun Context.shareText(text: String) {
     }
     ContextCompat.startActivity(this, Intent.createChooser(intent, null), null)
 }
+
+fun Context?.hasInternet(): Boolean =
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/utils/Extensions.kt:763]: This function needs the ACCESS_NETWORK_STATE permission, and the Core module doesn't have it.

If it means adding a new unused permission, I don't think adding unused code to the Core module is a good idea.

So I'll just remove this function.

File: app/src/main/java/com/infomaniak/drive/data/documentprovider/CloudStorageProvider.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/data/documentprovider/CloudStorageProvider.kt:589]: Just an oversight.

**Review by JorisBodin — CHANGES_REQUESTED** [review]
When you move from one drive to another it makes a copy and not a move

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**